src/database/keyvalue.rs

```diff
@@ -200,6 +200,29 @@ impl BatchOperations for Batch {
     impl_batch_operations!({}, process_delete_batch);
 }
 
+trait Helpers {
+    fn delete_spent_utxo_set(
+        &mut self,
+        deleted_utxos: &mut Vec<LocalUtxo>,
+        to_del: &[LocalUtxo],
+    ) -> Result<(), Error>;
+}
```

Comment:
This seems redundant to me. Why have another Trait when the actual delete call is just `self.del_utxo()`?

Reply:
The idea was to make the code shorter and more readable by abstraction the functionality of deleting a list of spent utxos.

Reply:
While I agree in general abstracting away makes the code shorter in one place.. But you also end up distributing chunks of logic around the codebase. Now theres one more extra place to look at while reading the deletion logic.. Which can make reading through actually more difficult.. 

Having all the business logic in a single place can also add to readability..  Here to me, it seems the trait is only taking away 2 lines of code which could have been in the main function itself.. And its only used for `keyvalue`..